core-bundle/src/EventListener/BackendPreviewRedirectListener.php

```diff
@@ -42,6 +42,6 @@ public function __invoke(RequestEvent $event): void
             return;
         }
 
-        $event->setResponse(new RedirectResponse($request->getSchemeAndHttpHost().$request->getPathInfo().(null !== ($qs = $request->server->get('QUERY_STRING')) ? '?'.$qs : '')));
+        $event->setResponse(new RedirectResponse($request->getSchemeAndHttpHost().$request->getBasePath().$request->getPathInfo().(null !== ($qs = $request->server->get('QUERY_STRING')) ? '?'.$qs : '')));
```

Comment:
Please add a unit tests for this so we don‘t run into regressions.

Also, I have searched the code base and found the following variants:

* `$request->getSchemeAndHttpHost().$request->getRequestUri()` (1 time)
* `$request->getSchemeAndHttpHost().$request->getPathInfo()` (2 times)
* `$request->getSchemeAndHttpHost().$request->getBaseUrl().$request->getPathInfo()` (3 times)
* `$request->getSchemeAndHttpHost().$request->getBasePath()` (1 time)

`$request->getSchemeAndHttpHost().$request->getBasePath().$request->getPathInfo()` does not occur in the code base at all, therefore I‘m wondering if it is correct?

Reply:
> therefore I‘m wondering if it is correct?

The change in this particular instance is correct, otherwise you will be redirected to the wrong URL if you open a back end preview URL without being logged in.

I'll check the other instances as well.

Reply:
How do I reproduce this?

Reply:
1. Log out of the back end.
2. Access a preview URL via localhost + subfolder, e.g. `https://localhost/your-contao-instance/public/preview.php/contao/preview?page=3`

You will be redirect to

```
https://localhost/contao/login?_hash=…&redirect=https%3A%2F%2Flocalhost%2Fc50%2Fpublic%2Fpreview.php%2Fcontao%2Fpreview%3Fpage%3D3
```

instead of

```
https://localhost/your-contao-instance/public/contao/login?_hash=…&redirect=https%3A%2F%2Flocalhost%2Fc50%2Fpublic%2Fpreview.php%2Fcontao%2Fpreview%3Fpage%3D3
```

Reply:
I was also able to reproduce this, and this change fixed it.

Simpler reproduction was just opening https://localhost/your-contao-instance/public/preview.php/contao (without the need to log out).